runtime/doc/lua.txt

```diff
@@ -3901,18 +3901,22 @@ Iter:flatten({depth})                                         *Iter:flatten()*
         (`Iter`)
 
 Iter:fold({init}, {f})                                           *Iter:fold()*
-    Folds ("reduces") an iterator into a single value.
+    Folds ("reduces") an iterator into a single value.         *Iter:reduce()*
 
     Examples: >lua
         -- Create a new table with only even values
-        local t = { a = 1, b = 2, c = 3, d = 4 }
-        local it = vim.iter(t)
-        it:filter(function(k, v) return v % 2 == 0 end)
-        it:fold({}, function(t, k, v)
-          t[k] = v
-          return t
-        end)
-        -- { b = 2, d = 4 }
+        vim.iter({ a = 1, b = 2, c = 3, d = 4 })
+          :filter(function(k, v) return v % 2 == 0 end)
+          :fold({}, function(t, k, v)
+            t[k] = v
+            return t
+          end) --> { b = 2, d = 4 }
```

Comment:
The shadowing of `t` added unnecessary questions to the old example. Also seems unnecessary to introduce intermediate variable like `it` for the iterator, instead of chaining methods. Chaining methods emphasizes the "pipeline" aspect of `vim.iter`.